astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,105 @@ def info(filename, output=None, **kwargs):
+            printdiff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+
+            printdiff('inA.fits', 'inB.fits', 'sci')
+            printdiff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive.
+
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        arguments or as a tuple::
+
+            printdiff('inA.fits', 'inB.fits', 'sci', 2)  # EXTNAME='SCI' & EXTVER=2
+            printdiff('inA.fits', 'inB.fits', extname='sci', extver=2)  # equivalent
+            printdiff('inA.fits', 'inB.fits', ('sci', 2))  # equivalent
+
+        Ambiguous or conflicting specifications will raise an exception::
+
+            printdiff('inA.fits', 'inB.fits', ext=('sci',1), extname='err', extver=2)
+
+    kwargs
+        Any additional keyword arguments to be passed to
+        `astropy.io.fits.FITSDiff`.
+
+    Notes
+    -----
+    The primary use for the `printdiff` function is to allow quick print out
+    of a FITS difference report and will write to ``sys.stdout``.
+    To save the diff report to a file please use `astropy.io.fits.FITSDiff`
+    directly.
+    """
+
+    # Pop extension keywords
+    extension = {key: kwargs.pop(key) for key in ['ext', 'extname', 'extver'] if key in kwargs}
+    has_extensions = len(args) != 0 or extension
+
+    if isinstance(inputa, string_types) and has_extensions:
+        # Use handy _getext to interpret any ext keywords
+        modea, closeda = _get_file_mode(inputa)
+        modeb, closedb = _get_file_mode(inputb)
+        hdulista, extidxa = _getext(inputa, modea, *args, **extension)
+        hdulistb, extidxb = _getext(inputa, modeb, *args, **extension)
+
+        try:
+            hdua = hdulista[extidxa]
+            hdub = hdulistb[extidxb]
+            # See above print for note
+            print(HDUDiff(hdua, hdub, **kwargs).report())
+
+        finally:
+            # See note about _close in getheader function
+            hdulista._close(closed=closeda)
+            hdulistb._close(closed=closedb)
+            return None
+
+    # If input is not a string, can feed HDU objects or HDUList directly,
+    # but can't currently handle extensions
+    elif isinstance(inputa, _ValidHDU) and has_extensions:
+            raise ValueError("Cannot use extension keywords when providing an HDU Object")
+
+    elif isinstance(inputa, _ValidHDU) and not has_extensions:
+            print(HDUDiff(inputa, inputb, **kwargs).report())
+            return None
+
+    elif isinstance(inputa, HDUList) and has_extensions:
+            raise NotImplementedError("Extension specification with HUDList Objects not implemented")
+
+    # This function is EXCLUSIVELY for printing the diff report to screen
+    # in a one-liner call, hence the use of print instead of logging
+    print(FITSDiff(inputa, inputb, **kwargs).report())
+
+
```

Comment:
@MSeifert04 🥇
@pllim 🥈  

But I was quite surprised how many comments you had that I missed. 😅